system/java_certs.py

```diff
@@ -0,0 +1,199 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# keytool -printcert -rfc -sslserver ppid.rsd-cloud.com:443 | keytool -importcert -keystore /usr/lib/jvm/java-7-oracle-amd64/jre/lib/security/cacerts -storepass changeit -alias ppid.rsd-cloud.com
+
+DOCUMENTATION = '''
+---
+module: java_cert
+version_added: "2.0"
+short_description: Uses keytool to import/remove key from java keystore(cacerts)
+description:
+  - This is a wrapper module around keytool. Which can be used to import/remove certificates from a given java keystore.
+options:
+  cert_url:
+    description:
+      - Basic URL to fetch SSL certificate from
+    required: true
+  cert_port:
+    description:
+      - Port to connect to URL. This will be used to create server URL:PORT
+    required: false
+    default: 443
+  cert_path:
+    description:
+      - Local path to load certificate from
+    required: false
+    default: null
+  keystore_path:
+    description:
+      - Path to keystore.
+    required: false
+    default: null
+  keystore_pass:
+    description:
+      - Keystore password
+    required: false
+    default: null
+  state:
+    description:
+      - Defines action which can be either certificate import or removal.
+    choices: [ 'present', 'absent' ]
+    default: 'present'
+    required: false
+
+author: Adam Hamsik
+'''
+
+EXAMPLES = '''
+# Import SSL certificate from google.com to a given cacerts keystore
+java_certs: cert_url=google.com cert_port=443 keystore_path=/usr/lib/jvm/jre7/lib/security/cacerts keystore_pass=changeit state=present
+
+# Remove certificate with given alias from a keystore
+java_certs: cert_url=google.com keystore_path=/usr/lib/jvm/jre7/lib/security/cacerts keystore_pass=changeit state=absent
+'''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import os
+import subprocess
+import sys
+import datetime
+import syslog
+
+def check_cert_present(module, keystore_path, keystore_pass, alias):
+    test_cmd = "keytool -noprompt -list -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+    (rc, out, err) = self.module.run_command(test_cmd)
+    if rc == 0:
+        return True
+    return False
+
+def import_cert_url(module, url, port, keystore_path, keystore_pass, alias):
+    fetch_cmd = "keytool -printcert -rfc -sslserver %s:%s" % (url, port)
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Fetch SSL certificate from remote host.
+    (rc, fetch_out, fetch_err) = module.run_command(fetch_cmd, check_rc=True)
+
+    # Use remote certificate from remote host and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, data=fetch_out, check_rc=False)
+    if not rc:
```

Comment:
Prefer `if rc==0:`

Not sure this section is required however as we test if key exists before calling this method.